src/components/ContextMenu/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useRef, useEffect, useLayoutEffect } from 'react';
+import styled from 'styled-components';
+import ContextMenuItem from './ContextMenuItem';
+
+function ContextMenu({ items, mousePos, parentRef, isToLeft, displayFocus }) {
+  const [isVisible, setIsVisible] = useState(false);
+  const [openToLeft, setOpenToLeft] = useState(false);
+  const [menuRef, setRefPos] = useState(null);
+  const [menuPos, setMenuPos] = useState({
+    top: '',
+    left: '',
+  });
+  const elementRef = useRef(null);
+
+  useLayoutEffect(() => {
+    if (displayFocus || parentRef)
+      setRefPos(elementRef.current.getBoundingClientRect());
+  }, []);
+
+  useEffect(() => {
+    initializeMenuPosition(mousePos, menuRef, parentRef);
+  }, [mousePos, menuRef]);
+
+  function initializeMenuPosition(mousePos, menuRef, parentRef) {
+    const windowWidth = window.innerWidth;
+    const windowHeight = window.innerHeight;
+    const newMenuPos = {
+      top: '',
+      left: '',
+    };
+
+    if (mousePos && menuRef) {
+      newMenuPos.top =
+        mousePos.y + menuRef.height > windowHeight
+          ? mousePos.y - menuRef.height
+          : mousePos.y;
+
+      newMenuPos.left =
+        mousePos.x + menuRef.width > windowWidth
+          ? mousePos.x - menuRef.width
+          : mousePos.x;
+
+      setMenuPos(prev => ({
+        ...prev,
```

Comment:
If `top` and `left` are the only props this state object ever has, that spread is unneeded

Reply:
Fixed.